src/snowflake/cli/api/project/schemas/native_app/path_mapping.py

```diff
@@ -1,10 +1,22 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import Any, Dict, List, Optional, Union
 
+from pydantic import Field
 from snowflake.cli.api.project.schemas.updatable_model import UpdatableModel
 
 
+class AnnotationProcessor(UpdatableModel):
+    name: str = Field(
+        title="Name of the processor to be invoked to discover the annotated code."
```

Comment:
nit: "Name of a processor to invoke on a collection of artifacts" ? Annotation processing is a special case.